lib/beaker/hypervisor/fusion.rb

```diff
@@ -54,6 +55,25 @@ def provision
         end
         time = Time.now - start
         @logger.notify "Spent %.2f seconds resuming VM" % time
+
+        begin
+          try_ssh_connection(host)
+        rescue *Beaker::SshConnection::RETRYABLE_EXCEPTIONS => e
+          @logger.notify "Obtaining IP information for #{host.name}"
+          ip = vm.network_info.data[vm.network_info.data.keys[0]]['ip_address']
+          if not ip
+            raise "Unable to connect to vm via hostname #{host.name}, and ip address is unavailable via vmname #{host[:vmname]}."
+          else
+            @logger.notify "Found IP address #{ip} for #{host.name}. Setting host.ip value"
+            host[:ip] = ip
+          end
+
+          # Set the hostname for each box
+          set_hostnames()
```

Comment:
Could this be generally useful?  Should it be the default case and not just the host-by-ip case?